ROX-15458: Implement the back-channel from Central to Sensor for Node-Inventories

vikin91 opened the pull request:
## Description

Central (node-inventory pipeline) can report back to Sensor (nodeInventoryHandlerImpl) a successful scanning of the NodeInventory.

Left for the follow-up PR: Sensor should handle the received ACK and forward it to Compliance for handling.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added - more will be added in [ROX-16861](https://issues.redhat.com/browse/ROX-16861) 

### N/A
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed

- Local deployment and looking at the logs
- CI

Maddosaurus reviewed (approved):
LGTM!
Only question I have: I vaguely remember we wanted to have the user-facing verbiage be "Node Scanning" instead of inventory. I'm personally okay with keeping the log messages in `pipeline.go` as they are, but maybe we should have them all consistent in their wording, right?

vikin91 commented:
> LGTM! Only question I have: I vaguely remember we wanted to have the user-facing verbiage be "Node Scanning" instead of inventory. I'm personally okay with keeping the log messages in `pipeline.go` as they are, but maybe we should have them all consistent in their wording, right?

Thanks! Yes, that is true. Actually the log messages here are more of us as for the customers, as the feature will work even without the ACKs. I will correct it on the next occasion - either in this PR or in the follow-up.

vikin91 commented on proto/internalapi/central/sensor_iservice.proto:
```diff
@@ -65,6 +65,18 @@ message MsgToSensor {
+
+    enum Action {
+        ACK = 0;
+        NACK = 1;
```
This NACK is between Central and Sensor, the other NACK that you mention is between Sensor and Compliance. We need both of them.

vikin91 commented:
Thank you for the reviews @jschnath and @Maddosaurus !
I fixed failing unit tests and added more unit test coverage in  [d322655](https://github.com/stackrox/stackrox/pull/6155/commits/d322655234f7cada341ae4f6aa599c29d1be0eb5)
